superduperdb/ext/utils.py

```diff
@@ -23,3 +29,22 @@ def format_prompt(X: str, prompt: str, context: t.Optional[t.List[str]] = None)
             raise ValueError(f'A context is required for prompt {prompt}')
     else:
         return prompt + X
+
+
+def superduperencode(object):
+    if isinstance(object, np.ndarray):
+        from superduperdb.ext.numpy import array
+
+        encoded = array(dtype=object.dtype, shape=object.shape).encode(object)
+        encoded['shape'] = object.shape
+        encoded['dtype'] = str(object.dtype)
+        return encoded
```

Comment:
extend this to other

Reply:
`r - Document`

`r['x'] - Encodable`

So bytes is `r['x'].encode()`